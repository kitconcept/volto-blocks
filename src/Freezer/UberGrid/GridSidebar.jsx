import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import withObjectBrowser from '@plone/volto/components/manage/Sidebar/ObjectBrowser';

import { Accordion, Segment } from 'semantic-ui-react';
import { FormattedMessage, injectIntl } from 'react-intl';
import { Icon } from '@plone/volto/components';

import ImageData from './ImageData';

import upSVG from '@plone/volto/icons/up-key.svg';
import downSVG from '@plone/volto/icons/down-key.svg';

const GridSidebar = (props) => {
  const { data, gridType } = props;
  const [activeAccIndex, setActiveAccIndex] = React.useState(0);

  function handleAccClick(e, titleProps) {
    const { index } = titleProps;
    const newIndex = activeAccIndex === index ? -1 : index;

    setActiveAccIndex(newIndex);
  }

  return (
    <Segment.Group raised>
      <header className="header pulled">
        <h2>
          {!gridType && <FormattedMessage id="Grid" defaultMessage="Grid" />}
          {gridType === 'images' && (
            <FormattedMessage id="Images Grid" defaultMessage="Images Grid" />
          )}
        </h2>
      </header>

      <Accordion fluid styled className="form">
        {data.columns &&
          data.columns.map((column, index) => (
            <>
              <Accordion.Title
                active={activeAccIndex === index}
                index={index}
                onClick={handleAccClick}
              >
                {!gridType && (
                  <FormattedMessage
                    id="Grid Element"
                    defaultMessage="Grid Element {index}"
                    values={{ index: index }}
                  />
                )}
                {gridType === 'images' && (
                  <FormattedMessage
                    id="Grid Image"
                    defaultMessage="Grid Image {index}"
                    values={{ index: index }}
                  />
                )}

                {activeAccIndex === 0 ? (
                  <Icon name={upSVG} size="20px" />
                ) : (
                  <Icon name={downSVG} size="20px" />
                )}
              </Accordion.Title>
              <Accordion.Content active={activeAccIndex === index}>
                <ImageData {...props} data={column} />
              </Accordion.Content>
            </>
          ))}
      </Accordion>
    </Segment.Group>
  );
};

GridSidebar.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  block: PropTypes.string.isRequired,
  onChangeBlock: PropTypes.func.isRequired,
  openObjectBrowser: PropTypes.func.isRequired,
};

export default compose(withObjectBrowser, injectIntl)(GridSidebar);
