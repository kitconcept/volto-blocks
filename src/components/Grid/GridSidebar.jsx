import React from 'react';
import PropTypes from 'prop-types';
import { Accordion, Button, Segment } from 'semantic-ui-react';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
import { Icon } from '@plone/volto/components';

import ImageData from './ImageData';
import TeaserData from './TeaserData';

import upSVG from '@plone/volto/icons/up-key.svg';
import downSVG from '@plone/volto/icons/down-key.svg';
import trashSVG from '@plone/volto/icons/delete.svg';
import addSVG from '@plone/volto/icons/add.svg';

const GridSidebar = props => {
  const { data, gridType } = props;
  const [activeAccIndex, setActiveAccIndex] = React.useState(0);

  function handleAccClick(e, titleProps) {
    const { index } = titleProps;
    const newIndex = activeAccIndex === index ? -1 : index;

    setActiveAccIndex(newIndex);
  }

  return (
    <Segment.Group raised>
      <header className="header pulled">
        <h2>
          {!gridType && <FormattedMessage id="Grid" defaultMessage="Grid" />}
          {gridType === 'image' && (
            <FormattedMessage id="Images Grid" defaultMessage="Images Grid" />
          )}
          {gridType === 'teaser' && (
            <FormattedMessage id="Teaser Grid" defaultMessage="Teaser Grid" />
          )}
        </h2>
        <Button.Group>
          <Button
            icon
            basic
            onClick={e => props.addNewColumn(e, gridType)}
            disabled={data.columns && data.columns.length >= 4}
          >
            <Icon name={addSVG} size="24px" />
          </Button>
        </Button.Group>
      </header>

      <Accordion fluid styled className="form">
        {data.columns &&
          data.columns.map((column, index) => (
            <React.Fragment key={column.id}>
              <Accordion.Title
                active={activeAccIndex === index}
                index={index}
                onClick={handleAccClick}
              >
                {(!gridType || gridType === 'teaser') && (
                  <>
                    <FormattedMessage
                      id="Grid Element"
                      defaultMessage="Grid Element {index}"
                      values={{ index: (index + 1).toString() }}
                    />
                  </>
                )}
                {gridType === 'image' && (
                  <FormattedMessage
                    id="Grid Image"
                    defaultMessage="Grid Image {index}"
                    values={{ index: (index + 1).toString() }}
                  />
                )}
                <div className="accordion-tools">
                  <Button.Group>
                    <Button
                      icon
                      basic
                      onClick={e => props.removeColumn(e, index)}
                    >
                      <Icon name={trashSVG} size="20px" color="#e40166" />
                    </Button>
                  </Button.Group>
                  {activeAccIndex === 0 ? (
                    <Icon name={upSVG} size="20px" />
                  ) : (
                    <Icon name={downSVG} size="20px" />
                  )}
                </div>
              </Accordion.Title>
              <Accordion.Content active={activeAccIndex === index}>
                {gridType === 'image' && (
                  <ImageData {...props} data={{ ...column, index }} />
                )}
                {gridType === 'teaser' && (
                  <TeaserData {...props} data={{ ...column, index }} />
                )}
              </Accordion.Content>
            </React.Fragment>
          ))}
      </Accordion>
    </Segment.Group>
  );
};

GridSidebar.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  tile: PropTypes.string.isRequired,
  onChangeTile: PropTypes.func.isRequired,
  openObjectBrowser: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
};

export default injectIntl(GridSidebar);
