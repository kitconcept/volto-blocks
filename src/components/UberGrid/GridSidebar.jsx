import React from 'react';
import PropTypes from 'prop-types';
import { Accordion, Button, Segment } from 'semantic-ui-react';
import { FormattedMessage, injectIntl } from 'react-intl';
import { Icon } from '@plone/volto/components';

import upSVG from '@plone/volto/icons/up-key.svg';
import downSVG from '@plone/volto/icons/down-key.svg';
import trashSVG from '@plone/volto/icons/delete.svg';
import addSVG from '@plone/volto/icons/add.svg';

const GridSidebar = (props) => {
  const {
    data,
    gridType,
    sidebarData,
    activeColumn,
    onChangeSelectedColumnItem,
  } = props;

  function handleChangeColumn(e, blockProps) {
    const { index } = blockProps;
    const newIndex = activeColumn === index ? -1 : index;

    onChangeSelectedColumnItem(newIndex);
  }

  return (
    <Segment.Group raised>
      <header className="header pulled">
        <h2>
          {!gridType && <FormattedMessage id="Grid" defaultMessage="Grid" />}
          {gridType === 'image' && (
            <FormattedMessage id="Images Grid" defaultMessage="Images Grid" />
          )}
          {gridType === 'teaser' && (
            <FormattedMessage id="Teaser Grid" defaultMessage="Teaser Grid" />
          )}
          {gridType === 'listing' && (
            <FormattedMessage id="Listing Grid" defaultMessage="Listing Grid" />
          )}
        </h2>
        <Button.Group>
          <Button
            icon
            basic
            onClick={(e) => props.addNewColumn(e, gridType)}
            disabled={data.columns && data.columns.length >= 4}
          >
            <Icon name={addSVG} size="24px" />
          </Button>
        </Button.Group>
      </header>
      hey
      <Accordion fluid styled className="form">
        {data.columns &&
          data.columns.map((column, index) => (
            <React.Fragment key={column.id}>
              <Accordion.Title
                active={activeColumn === index}
                index={index}
                onClick={handleChangeColumn}
              >
                {(!gridType || gridType === 'teaser') && (
                  <>
                    <FormattedMessage
                      id="Grid Element"
                      defaultMessage="Grid Element {index}"
                      values={{ index: (index + 1).toString() }}
                    />
                  </>
                )}
                {gridType === 'image' && (
                  <FormattedMessage
                    id="Grid Image"
                    defaultMessage="Grid Image {index}"
                    values={{ index: (index + 1).toString() }}
                  />
                )}
                {gridType === 'listing' && (
                  <FormattedMessage
                    id="ListingGrid"
                    defaultMessage="Listing Grid {index}"
                    values={{ index: (index + 1).toString() }}
                  />
                )}
                <div className="accordion-tools">
                  {data.columns.length > 2 && (
                    <Button.Group>
                      <Button
                        icon
                        basic
                        onClick={(e) => props.removeColumn(e, index)}
                      >
                        <Icon name={trashSVG} size="20px" color="#e40166" />
                      </Button>
                    </Button.Group>
                  )}
                  {activeColumn === 0 ? (
                    <Icon name={upSVG} size="20px" />
                  ) : (
                    <Icon name={downSVG} size="20px" />
                  )}
                </div>
              </Accordion.Title>
              <Accordion.Content active={activeColumn === index}>
                {/* {sidebarData(props, column, index)} */}
              </Accordion.Content>
            </React.Fragment>
          ))}
      </Accordion>
    </Segment.Group>
  );
};

GridSidebar.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  block: PropTypes.string.isRequired,
  onChangeBlock: PropTypes.func.isRequired,
  sidebarData: PropTypes.func.isRequired,
};

export default injectIntl(GridSidebar);
