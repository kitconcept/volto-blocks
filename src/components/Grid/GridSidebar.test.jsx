import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import GridSidebar from './GridSidebar';

const mockStore = configureStore();

test('renders a Grid Block Sidebar component', () => {
  const store = mockStore({
    content: {
      create: {},
      data: {},
    },
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <GridSidebar
        id="dcdf1f42-645d-48f6-9531-357bdc2e1881"
        data={{
          '@type': 'teaserGrid',
          columns: [
            {
              '@type': 'teaser',
              href: '/news/a-news-item-that-tells-things',
              id: '2110a241-1389-4cda-8811-77031a540efa',
              index: 0,
            },
            {
              '@type': 'teaser',
              href: '/news/another-news-item',
              id: '3ccd6338-4d13-4f22-a68c-14b6b60cf860',
              index: 1,
            },
            {
              '@type': 'teaser',
              id: '2ac3b8d1-3ea8-4d78-a4ce-b2997a961c5e',
            },
            {
              '@type': 'teaser',
              id: 'e947b1e5-b9b1-4fc1-acf3-cb04f78a3574',
            },
          ],
        }}
        type="teaserGrid"
        block="1234"
        pathname="/news"
        onChangeBlock={() => {}}
        openObjectBrowser={() => {}}
        sidebarData={() => <div />}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
