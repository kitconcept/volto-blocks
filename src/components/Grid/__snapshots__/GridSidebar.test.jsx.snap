// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Grid Tile Sidebar component 1`] = `
<div
  className="ui raised segments"
>
  <header
    className="header pulled"
  >
    <h2>
      Grid
    </h2>
    <div
      className="ui buttons"
    >
      <button
        className="ui basic icon disabled button"
        disabled={true}
        onClick={[Function]}
        tabIndex={-1}
      >
        <svg
          className="icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          onClick={null}
          style={
            Object {
              "fill": "currentColor",
              "height": "24px",
              "width": "auto",
            }
          }
          viewBox=""
          xmlns=""
        />
      </button>
    </div>
  </header>
  <div
    className="accordion ui fluid styled form"
  >
    <div
      className="active title"
      onClick={[Function]}
    >
      Grid Element 1
      <div
        className="accordion-tools"
      >
        <div
          className="ui buttons"
        >
          <button
            className="ui basic icon button"
            onClick={[Function]}
          >
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "#e40166",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </button>
        </div>
        <svg
          className="icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          onClick={null}
          style={
            Object {
              "fill": "currentColor",
              "height": "20px",
              "width": "auto",
            }
          }
          viewBox=""
          xmlns=""
        />
      </div>
    </div>
    <div
      className="content active"
    >
      <div />
    </div>
    <div
      className="title"
      onClick={[Function]}
    >
      Grid Element 2
      <div
        className="accordion-tools"
      >
        <div
          className="ui buttons"
        >
          <button
            className="ui basic icon button"
            onClick={[Function]}
          >
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "#e40166",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </button>
        </div>
        <svg
          className="icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          onClick={null}
          style={
            Object {
              "fill": "currentColor",
              "height": "20px",
              "width": "auto",
            }
          }
          viewBox=""
          xmlns=""
        />
      </div>
    </div>
    <div
      className="content"
    >
      <div />
    </div>
    <div
      className="title"
      onClick={[Function]}
    >
      Grid Element 3
      <div
        className="accordion-tools"
      >
        <div
          className="ui buttons"
        >
          <button
            className="ui basic icon button"
            onClick={[Function]}
          >
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "#e40166",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </button>
        </div>
        <svg
          className="icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          onClick={null}
          style={
            Object {
              "fill": "currentColor",
              "height": "20px",
              "width": "auto",
            }
          }
          viewBox=""
          xmlns=""
        />
      </div>
    </div>
    <div
      className="content"
    >
      <div />
    </div>
    <div
      className="title"
      onClick={[Function]}
    >
      Grid Element 4
      <div
        className="accordion-tools"
      >
        <div
          className="ui buttons"
        >
          <button
            className="ui basic icon button"
            onClick={[Function]}
          >
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "#e40166",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </button>
        </div>
        <svg
          className="icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          onClick={null}
          style={
            Object {
              "fill": "currentColor",
              "height": "20px",
              "width": "auto",
            }
          }
          viewBox=""
          xmlns=""
        />
      </div>
    </div>
    <div
      className="content"
    >
      <div />
    </div>
  </div>
</div>
`;
